lldb/source/Host/CMakeLists.txt

```diff
@@ -108,13 +108,9 @@ else()
       linux/HostInfoLinux.cpp
       linux/LibcGlue.cpp
       linux/Support.cpp
+      android/HostInfoAndroid.cpp
+      android/LibcGlue.cpp
       )
-    if (CMAKE_SYSTEM_NAME MATCHES "Android")
```

Comment:
What is wrong with android detection? I doubt this is a proper solution

Reply:
I use "Linux" to do canadian compilation from linux to android without using android NDK. That is why this is not correct.

Reply:
Well, so this is not a cmake problem then.

Reply:
> Well, so this is not a cmake problem then.

but it still needs a solution tbh. android is linux.

Reply:
Not quite. It's distant enough in many aspects (from the toolchain perspective) to have a separate cmake system name.

Reply:
That is probably because cmake treats Android differently, despite it can be treated the same.

Reply:
I will try to find a better solution.

Reply:
While Android might be Linux, the inverse isn't true. Amongst other things, this change would result in all Linux hosts reporting that they support Android triples.